src/components/data_update/DatasetUpdate.js

```diff
@@ -0,0 +1,392 @@
+
+  const searchOntologies = async (query, callback) => {
+    let options = [];
+    await DAR.getAutoCompleteOT(query).then(
+      items => {
+        options = items.map((item) => {
+          return item.label;
+        });
+        callback(options);
+      });
+  };
+
+  const dacOptions = (dacs) => {
+    let options = [];
+    if (!isNil(dacs)) {
+      options = dacs.map((dac) => {
+        return { displayText: dac.name, dacId: dac.dacId };
+      });
+    }
+    return options;
+  };
+
+  const getDiseaseLabels = async (ontologyIds) => {
+    let labels = [];
+    if (!isNil(ontologyIds)) {
+      const idList = ontologyIds.join(',');
+      const result = await DAR.searchOntologyIdList(idList);
+      labels = result.map(r => r.label);
+    }
+    return labels;
+  };
+
+  const extract = useCallback((propertyName) => {
+    const property = find({ propertyName })(dataset.properties);
+    return property?.propertyValue;
+  }, [dataset]);
+
+  const asProperty = (propertyName, propertyValue) => {
+    return {
+      propertyName,
+      propertyValue
+    };
+  };
+
+  const normalizeDataUse = useCallback(async (dataUse) => {
+    let du = dataUse;
+    if (!isNil(dataUse.diseaseRestrictions)) {
+      du.hasDiseaseRestrictions = true;
+      du.diseaseLabels = await getDiseaseLabels(dataUse.diseaseRestrictions);
+    }
+    if (!isNil(dataUse.other)) {
+      du.hasPrimaryOther = true;
+    }
+    if (!isNil(dataUse.secondaryOther)) {
+      du.hasSecondaryOther = true;
+    }
+    return du;
+  }, []);
+
+  const submitForm = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const formElement = event.target.form;
+    const submitData = new FormData(formElement);
+    const nihFileData = submitData.get('nihInstitutionalCertificationFile');
+    const consentGroups = [{ nihInstitutionalCertificationFile: nihFileData }];
+
+    const newDataset = {
+      name: dataset.name,
+      dacId: formData.dac.dacId,
+      properties: [
+        asProperty('Dataset Name', formData.properties.datasetName),
+        asProperty('Data Type', formData.properties.dataType),
+        asProperty('Species', formData.properties.species),
+        asProperty('Phenotype/Indication', formData.properties.phenotype),
+        asProperty('# of participants', formData.properties.nrParticipants),
+        asProperty('Description', formData.properties.description),
+        asProperty('dbGAP', formData.properties.dbGap),
+        asProperty('Data Depositor', formData.properties.dataDepositor),
+        asProperty('Principal Investigator(PI)', formData.properties.principalInvestigator),
+      ]
+    };
+
+    const multiPartFormData = new FormData();
+    multiPartFormData.append('dataset', JSON.stringify(newDataset));
+    multiPartFormData.append('consentGroups', consentGroups);
+
+    DataSet.updateDatasetV3(dataset.dataSetId, multiPartFormData).catch(() => {
```

Comment:
Additionally, although the response returns 200, the underlying dataset does not actually get updated. I think there's probably an issue with the `updateDataset` endpoint.

Reply:
I'm seeing some updates when using the submit button. For example, the dataset name updates, but the species doesn't seem to update. I think the properties are not saving correctly. What I think is missing is some UI indication of success.

Reply:
Just like the submission form, I added a success notification and a redirect to the catalog page.